Sources/OpenTelemetryApi/Trace/Span.swift

```diff
@@ -111,6 +111,40 @@ public extension Span {
     func setAttribute(key: SemanticAttributes, value: Bool) {
         return setAttribute(key: key.rawValue, value: AttributeValue.bool(value))
     }
+
```

Comment:
This fundamentally change the context by running actions in a block, meaning context association cannot be made using the current method on iOS. We'd prefer this to be in a separate `#ifdef` for linux, to avoid users from mixing these APIs.

Reply:
I'm not sure I follow why this wouldn't work on iOS? Could you elaborate a bit?